pkg/securelaunch/helpers.go

```diff
@@ -48,71 +67,232 @@ func WriteToFile(data []byte, dst, defFileName string) (string, error) {
 		Debug("New target=%s", target)
 	}
 
-	Debug("target=%s", target)
+	Debug("WriteToFile: target=%s", target)
 	err = ioutil.WriteFile(target, data, 0644)
 	if err != nil {
 		return "", fmt.Errorf("failed to write date to file =%s, err=%v", target, err)
 	}
-	Debug("WriteToFile exit w success data written to target=%s", target)
+	Debug("WriteToFile: exit w success data written to target=%s", target)
 	return target, nil
 }
 
-/*
- * GetMountedFilePath returns a file path corresponding to a <device_identifier>:<path> user input format.
- * <device_identifier> may be a Linux block device identifier like sda or a FS UUID.
- *
- * NOTE: Caller's responsbility to unmount this..use return var mountPath to unmount in caller.
- */
-func GetMountedFilePath(inputVal string, flags uintptr) (string, string, error) {
-	s := strings.Split(inputVal, ":")
-	if len(s) != 2 {
-		return "", "", fmt.Errorf("%s: Usage: <block device identifier>:<path>", inputVal)
+// persist writes data to targetPath.
+// targetPath is of form sda:/boot/cpuid.txt
+func persist(data []byte, targetPath string, defaultFile string) error {
+
+	filePath, r := GetMountedFilePath(targetPath, 0) // 0 is flag for rw mount option
```

Comment:
I would recommend using the constants from `os` (https://pkg.go.dev/os?tab=doc#pkg-constants)

Reply:
This flag i.e 0 is passed to device.Mount since this is a mount flag... where as the os flag is used for opening files... can these be used interchangably ?

Reply:
These are Mount flag options . Can these be used interchangeably with os constants ?... I doubt... I know I have hard coded 0 because I couldn’t find the equivalent Mount flag for unix.Write Mount...